docs/template-resolution.md

```diff
@@ -0,0 +1,45 @@
+# Template resolution strategies
```

Comment:
Please go over this file and update it in accordance with the implementation. Then link this file from the main README.md as the place where one can read more about the templating config.